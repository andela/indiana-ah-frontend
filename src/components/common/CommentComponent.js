import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CommentIcon from '../../images/icons/chat-1.png';

const CommentComponent = ({ commentCount }) => (
<Fragment>
<span className = 'icon_holder'>
<img src = {CommentIcon} alt = 'CommentIcon' className= 'icon_height icon_margin'/>
<sub className='count'>{commentCount}</sub>
</span>
</Fragment>
);

CommentComponent.propTypes = {
  commentCount: PropTypes.number.isRequired,
};

export default CommentComponent;
