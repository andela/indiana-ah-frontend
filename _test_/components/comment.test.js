/* eslint-disable no-unused-expressions */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CommentComponent from '../../src/components/common/CommentComponent';

describe('<CommentComponent />', () => {
  const wrapper = shallow(<CommentComponent commentCount={5} />);
  it('should render a comment component', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('span')).toHaveLength(1);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('sub')).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<CommentComponent commentCount={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
