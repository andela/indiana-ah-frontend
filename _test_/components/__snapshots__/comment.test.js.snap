// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentComponent /> renders correctly 1`] = `
<span
  className="icon_holder"
>
  <img
    alt="CommentIcon"
    className="icon_height icon_margin"
    src="test-file-stub"
  />
  <sub
    className="count"
  >
    5
  </sub>
</span>
`;
